### PR TITLE
Add docments for the TransformBlock section

### PR DESCRIPTION
**Pull request by butchland**
Adding docments for the transform block section
* TransformBlock
* CategoryBlock
* MultiCategoryBlock
* RegressionBlock

Ref #11

### CODE REVIEW

File: fastai/data/block.py [hunk 1]
```diff
@@ -12,25 +12,41 @@
...
 class TransformBlock():
     "A basic wrapper that links defaults transforms for the data block API"
-    def __init__(self, type_tfms=None, item_tfms=None, batch_tfms=None, dl_type=None, dls_kwargs=None):
+    def __init__(self,
+                 type_tfms:list=None, # List of type transforms
+                 item_tfms:list=None, # List of item transforms
+                 batch_tfms:list=None, # List of batch transforms
+                 dl_type:TfmdDL=None, # `TfmdDL` Dataloader class
+                 dls_kwargs:dict=None  # Additional arguments to be passed to `TfmdDL`
+    ):
```
Comment (muellerzr) [inline fastai/data/block.py:21]: Fix the tabs (see the style guide)

```suggestion
    def __init__(self,
                 type_tfms:list=None, # List of type `Transform`'s
                 item_tfms:list=None, # List of `ItemTransform`'s, applied on an item
                 batch_tfms:list=None, # List of `Transform`'s or `RandTransforms`, applied by batch
                 dl_type:TfmdDL=None, # Specific class type that inherits `TfmdDL`
                 dls_kwargs:dict=None  # Additional arguments to be passed to `TfmdDL`
    ):
```
Questioning if we should use `dls_kwargs`, for now that's fine https://github.com/fastai/fastai/issues/3586

File: fastai/data/block.py [hunk 2]
```diff
@@ -12,25 +12,41 @@
...
         self.item_tfms  = ToTensor + L(item_tfms)
         self.batch_tfms =            L(batch_tfms)
         self.dl_type,self.dls_kwargs = dl_type,({} if dls_kwargs is None else dls_kwargs)
 
 # Cell
-def CategoryBlock(vocab=None, sort=True, add_na=False):
+def CategoryBlock(
+    vocab:(list, pd.Series)=None, # List of unique values, or a pandas series with categorical or unique values
+    sort:bool=True,  # Sort the category values?
+    add_na:bool=False # Add NA as a category?
```
Comment (muellerzr) [inline fastai/data/block.py:31]:
```suggestion
    vocab:list=None, # A list of unique class names
    sort:bool=True,  # Sort the classes alphabetically
    add_na:bool=False # Add `#na#` to `vocab`
```

File: fastai/data/block.py [hunk 3]
```diff
@@ -12,25 +12,41 @@
...
+):
     "`TransformBlock` for single-label categorical targets"
     return TransformBlock(type_tfms=Categorize(vocab=vocab, sort=sort, add_na=add_na))
 
 # Cell
-def MultiCategoryBlock(encoded=False, vocab=None, add_na=False):
+def MultiCategoryBlock(
+    encoded:bool=False, # Use one-hot encoding?
+    vocab:(list,pd.Series)=None, # List of unique values, or a pandas series with categorical or unique values
+    add_na:bool=False # Add NA as a category?
```
Comment (muellerzr) [inline fastai/data/block.py:40]: Same comment as `CategoryBlock`

File: fastai/data/block.py [hunk 4]
```diff
@@ -12,25 +12,41 @@
...
+    sort:bool=True,  # Sort the category values?
+    add_na:bool=False # Add NA as a category?
+):
     "`TransformBlock` for single-label categorical targets"
     return TransformBlock(type_tfms=Categorize(vocab=vocab, sort=sort, add_na=add_na))
 
 # Cell
-def MultiCategoryBlock(encoded=False, vocab=None, add_na=False):
+def MultiCategoryBlock(
+    encoded:bool=False, # Use one-hot encoding?
```
Comment (muellerzr) [inline fastai/data/block.py:38]:
```suggestion
    encoded:bool=False, # Whether the data comes in one-hot encoded
```

**Review by muellerzr — CHANGES_REQUESTED** [review]
Great start, left some notes and recommendations

File: fastai/data/block.py (hunk 3 above)
Comment (butchland) [reply fastai/data/block.py]: Accepted your suggestions, but actually using a series doesn't have to be a series of unique values, it just needs to be a series containing unique or categorical values, the Categorize transform will extract the unique values/categories

File: fastai/data/block.py [hunk 5]
```diff
@@ -12,25 +12,41 @@
...
 class TransformBlock():
     "A basic wrapper that links defaults transforms for the data block API"
-    def __init__(self, type_tfms=None, item_tfms=None, batch_tfms=None, dl_type=None, dls_kwargs=None):
+    def __init__(self,
+                 type_tfms:list=None, # List of `Transform`s
+                 item_tfms:list=None, # List of `ItemTransform`s, applied on an item
+                 batch_tfms:list=None, # List of `Transform`s or `RandTransform`s, applied by batch
+                 dl_type:TfmdDL=None, #  Specific class that inherits `TfmdDL`
+                 dls_kwargs:dict=None  # Additional arguments to be passed to `TfmdDL`
+    ):
```
Comment (warner-benjamin) [inline fastai/data/block.py:21]:
```suggestion
    def __init__(self,
        type_tfms:list=None, # List of `Transform`s
        item_tfms:list=None, # List of `ItemTransform`s, applied on an item
        batch_tfms:list=None, # List of `Transform`s or `RandTransform`s, applied by batch
        dl_type:TfmdDL=None, # Specific class that inherits `TfmdDL`
        dls_kwargs:dict=None # Additional arguments to be passed to `TfmdDL`
    ):
```

**Review by warner-benjamin — CHANGES_REQUESTED** [review]
There are still some [style](https://github.com/muellerzr/fastai-docment-sprint/blob/master/docs_src/style-guide.md) issues. I fixed one of them with the suggestion, but there are still some extra spaces in the other blocks which need to be removed.

File: fastai/data/block.py [hunk 6]
```diff
@@ -12,25 +12,41 @@
 # Cell
 class TransformBlock():
     "A basic wrapper that links defaults transforms for the data block API"
-    def __init__(self, type_tfms=None, item_tfms=None, batch_tfms=None, dl_type=None, dls_kwargs=None):
+    def __init__(self,
+        type_tfms:list=None, # List of `Transform`s
+        item_tfms:list=None, # List of `ItemTransform`s, applied on an item
+        batch_tfms:list=None, # List of `Transform`s or `RandTransform`s, applied by batch
+        dl_type:TfmdDL=None, # Specific class that inherits `TfmdDL`
+        dls_kwargs:dict=None, # Additional arguments to be passed to `TfmdDL`
```
Comment (muellerzr) [inline fastai/data/block.py:20]:
```suggestion
        dls_kwargs:dict=None, # Additional arguments to be passed to `DataLoaders`
```

**Review by muellerzr — APPROVED** [review]
One small nit then I approve

**Review by warner-benjamin — APPROVED** [review]
Looks good. Thanks!

**Pull request merged**